### PR TITLE
Do not use 'python _.py' for env

### PR DESCRIPTION
**Pull request by saschanaz**
CMake sets CMAKE_C_COMPILER as `python` when `env['CC']` is `python emcc.py`. Some projects including LLVM does `COMMAND ${CMAKE_C_COMPILER} -Wl,--version` and this causes hang because it runs Python REPL instead of emcc.

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -1384,6 +1384,12 @@ def remove_quotes(arg):
 
   @staticmethod
   def get_building_env(native=False, doublequote_commands=False):
+    # CMake does not like 'CC=python emcc.py', CMAKE_C_COMPILER becomes just 'python' without emcc
+    def shellpath(pypath):
+      ext = '.bat' if WINDOWS else ''
+      if pypath.endswith('.py'):
+        pypath = pypath[:-3]
```
Comment (kripken) [inline tools/shared.py]: why remove the suffix?

Comment (saschanaz) [reply tools/shared.py]: CMake splits `python emcc.py` into `python` and thinks `python` is the C compiler. I think this forces us to roll back to `emcc` from `python emcc.py` at least when setting environment variables.

**Comment by saschanaz** [comment]
BTW the nondeterminism random failures again... I neither can repro locally.

**Comment by kripken** [comment]
Looks good to me. Maybe @juj should take a look at this stuff too though.

(The nondeterministic failures should be fixed by https://github.com/WebAssembly/binaryen/pull/1360 )

**Comment by juj** [comment]
It would be great to have a test for this, since the failure sounds like something that would have been present before the earlier refactor already? Having env. var `CC` be "C:/path/to/emcc.bat` instead of `python C:/path/to/emcc.py` on Windows sounds ok (and might even be better in some cases), so this looks good to me, although highlighting the CMake issue in our tests would be great if you have a test case at hand, since this is something that hasn't been surfaced before.

**Comment by saschanaz** [comment]
I was using emmake, it seems I should use emcmake instead. emcmake explicitly sets CMAKE_C_COMPILER so no problem.

**Comment by kripken** [comment]
I actually wasn't aware that emcmake does something different. We should check if that's documented or not. Or, perhaps emmake could also set the cmake env vars? @juj

**Comment by juj** [comment]
> I actually wasn't aware that emcmake does something different.

`emcmake` is identical to `emconfigure`, there was someone who wanted to duplicate those two to make it feel more documentative. In general `emmake` is like `make`, and `emcmake` is like `cmake`, so `(em)cmake` is used to generate the build Makefiles, and `(em)make` is then used to build.

**Comment by juj** [comment]
Oh also, when generating Makefiles with `emcmake`, one can use regular `make` to build them. This is because the configuration gets "bundled" into the generated Makefiles by CMake, so any environment present during the invocation of `emcmake` should carry over to the regular `make` call.

**Pull request closed without merging**